src/Confluent.Kafka/Consumer.cs

```diff
@@ -414,6 +414,28 @@ public void Unassign()
             => consumer.Unassign();
 
         /// <summary>
+        ///     Store offsets for one or more partitions.
+        ///     
+        ///     The offset will be committed (written) to the offset store according
+        ///     to `auto.commit.interval.ms` or manual offset-less commit().
+        /// </summary>
+        /// <remarks>
+        ///     `enable.auto.offset.store` must be set to "false" when using this API.
+        /// </remarks>
+        /// <param name="message">
+        ///     The message used to determine partition of the offset.
+        /// </param>
+        /// <returns>
+        ///     Offset results with per-partition errors.
+        /// </returns>
+        public OffsetResults StoreOffset(Message<TKey, TValue> message)
```

Comment:
I don't think we should have an `OffsetResults class` - just return `List<TopicPartitionErrorOffset>` and throw an exception on global error.